internal/testplanet/uplink.go

```diff
@@ -321,3 +327,75 @@ func atLeastOne(value int) int {
 	}
 	return value
 }
+
+// GetMetainfo returns a metainfo and streams store for the given configuration and identity.
+func GetMetainfo(ctx context.Context, config uplink.Config, identity *identity.FullIdentity) (db storj.Metainfo, ss streams.Store, err error) {
+	tlsOpts, err := tlsopts.NewOptions(identity, config.TLS)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// ToDo: Handle Versioning for Uplinks here
+
+	tc := transport.NewClientWithTimeouts(tlsOpts, transport.Timeouts{
+		Request: config.Client.RequestTimeout,
+		Dial:    config.Client.DialTimeout,
+	})
+
+	if config.Client.SatelliteAddr == "" {
+		return nil, nil, errs.New("satellite address not specified")
+	}
+
+	m, err := metainfo.Dial(ctx, tc, config.Client.SatelliteAddr, config.Client.APIKey)
+	if err != nil {
+		return nil, nil, errs.New("failed to connect to metainfo service: %v", err)
+	}
+	defer func() {
+		if err != nil {
+			err = errs.Combine(err, m.Close())
+		}
+	}()
+
+	project, err := kvmetainfo.SetupProject(m)
+	if err != nil {
+		return nil, nil, errs.New("failed to create project: %v", err)
+	}
+
+	ec := ecclient.NewClient(tc, config.RS.MaxBufferMem.Int())
+	fc, err := infectious.NewFEC(config.RS.MinThreshold, config.RS.MaxThreshold)
+	if err != nil {
+		return nil, nil, errs.New("failed to create erasure coding client: %v", err)
+	}
+	rs, err := eestream.NewRedundancyStrategy(eestream.NewRSScheme(fc, config.RS.ErasureShareSize.Int()), config.RS.RepairThreshold, config.RS.SuccessThreshold)
+	if err != nil {
+		return nil, nil, errs.New("failed to create redundancy strategy: %v", err)
+	}
+
+	maxEncryptedSegmentSize, err := encryption.CalcEncryptedSize(config.Client.SegmentSize.Int64(), config.GetEncryptionScheme())
+	if err != nil {
+		return nil, nil, errs.New("failed to calculate max encrypted segment size: %v", err)
+	}
+	segment := segments.NewSegmentStore(m, ec, rs, config.Client.MaxInlineSize.Int(), maxEncryptedSegmentSize)
+
+	blockSize := config.GetEncryptionScheme().BlockSize
+	if int(blockSize)%config.RS.ErasureShareSize.Int()*config.RS.MinThreshold != 0 {
+		err = errs.New("EncryptionBlockSize must be a multiple of ErasureShareSize * RS MinThreshold")
+		return nil, nil, err
+	}
+
+	// TODO(jeff): there's some cycles with libuplink and this package in the libuplink tests
+	// and so this package can't import libuplink. that's why this function is duplicated
+	// in some spots.
```

Comment:
There shouldn't be import cycles when:
1. only test packages use testplanet
2. tests are written using `package xxx_test`

But I guess resolving the deps would be too much here.

Reply:
The comment is also wrong. It's not actually duplicated. I thought it would have to be, but it turns out only `testplanet` and tests that use `testplanet` ever call this.